common/TextDifferencer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startup.common;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of text difference based on the Longest Common Subsequence problem (A.K.A LCS).
+ */
+public class TextDifferencer {
+
+  /**
+   * Returns all the text differences between two given strings.
+   *
+   * @param first The first string.
+   * @param second The second string.
+   * @return A list which holds all the text differences.
+   */
+  public static List<CharDifference> getAllTextDifferences(String first, String second) {
+    return getDifferencesFromLCSMatrix(
+            computeLCSMatrix(first.toCharArray(), second.toCharArray()),
+            first.toCharArray(),
+            second.toCharArray())
+        .stream()
+        .map((builder) -> builder.build())
+        .collect(Collectors.toList());
+  }
+
+  /** Create an empty matrix based on the given dimensions. */
+  private static int[][] createEmptyLCSMatrix(int rowSize, int colSize) {
+    final int[][] lcsMatrix = new int[rowSize][];
+    for (int i = 0; i < lcsMatrix.length; ++i) {
```

Comment:
Let's use i++
Also below..

There's no difference in this context, and it's more standard.